File: Api/CustomerSupportAPI/CustomerSupportAPI.Domain/CustomerSupportModel.cs
namespace CustomerSupportAPI.Domain
{
    public class CustomerSupportModel
    {

        #region Properties
        public int Id { get; private set; }
        public string Email { get; private set; }
        public string Phone { get; private set; }
        public string? Number { get; private set; }
        public int TypeInquiry { get; private set; }
        public string Description { get; private set; }
        public bool AgreementTerms { get; private set; }

        #endregion Properties

        #region Constructors
        public CustomerSupportModel() { }
        public CustomerSupportModel(string email, string phone, string? number, int typeInquiry, string description, bool agreementTerms)
        {
            Email = email;
            Phone = phone;
            Number = number;
            TypeInquiry = typeInquiry;
            Description = description;
            AgreementTerms = agreementTerms;
        }
        public CustomerSupportModel(int id, string email, string phone, string? number, int typeInquiry, string description, bool agreementTerms)
        {
            Id = id;
            Email = email;
            Phone = phone;
            Number = number;
            TypeInquiry = typeInquiry;
            Description = description;
            AgreementTerms = agreementTerms;
        }

        #endregion Constructors

        #region Methods
        public void UpdateEntity(int id, string email, string phone, string? number, int typeInquiry, string description, bool agreementTerms)
        {
            Id = id;
            Email = email;
            Phone = phone;
            Number = number;
            TypeInquiry = typeInquiry;
            Description = description;
            AgreementTerms = agreementTerms;
        }
        #endregion Methods
    }
}

File: Web/src/app/models/customer-support.model.ts
export class CustomerSupportModel {

    email!: string;
    phone!: string;
    number!: string;
    typeInquiry!: number;
    description!: string;
    agreementTerms!: boolean;

    constructor(email: string, phone: string, number: string, typeInquiry: number, description: string, terms: boolean) {
        this.email = email;
        this.phone = phone;
        this.number = number;
        this.typeInquiry = +typeInquiry;
        this.description = description;
        this.agreementTerms = terms;
    }
}
